base_layer/mmr/src/common.rs

```diff
@@ -214,14 +217,14 @@ mod test {
 
     #[test]
     fn leaf_to_node_indices() {
-        assert_eq!(node_index(0), 0);
-        assert_eq!(node_index(1), 1);
-        assert_eq!(node_index(2), 3);
-        assert_eq!(node_index(3), 4);
-        assert_eq!(node_index(5), 8);
-        assert_eq!(node_index(6), 10);
-        assert_eq!(node_index(7), 11);
-        assert_eq!(node_index(8), 15);
+        assert_eq!(node_index(LeafIndex(0)), 0);
+        assert_eq!(node_index(LeafIndex(1)), 1);
+        assert_eq!(node_index(LeafIndex(2)), 3);
+        assert_eq!(node_index(LeafIndex(3)), 4);
+        assert_eq!(node_index(LeafIndex(5)), 8);
+        assert_eq!(node_index(LeafIndex(6)), 10);
+        assert_eq!(node_index(LeafIndex(7)), 11);
+        assert_eq!(node_index(LeafIndex(8)), 15);
```

Comment:
why not implement `From` and then you can just do 

```suggestion
        assert_eq!(node_index(8.into()), 15);
```

Reply:
I thought about it. But then again, someone will take a number and without thinking call an `into` on it. And we will have the same hard to spot bug mixing node index and leaf index.